proxy/src/main/java/com/velocitypowered/proxy/connection/client/HandshakeSessionHandler.java

```diff
@@ -114,6 +114,10 @@ private ConnectionType checkForForge(Handshake handshake) {
     if (handshake.getServerAddress().endsWith(LegacyForgeConstants.HANDSHAKE_HOSTNAME_TOKEN)
         && handshake.getProtocolVersion().compareTo(ProtocolVersion.MINECRAFT_1_13) < 0) {
       return ConnectionTypes.LEGACY_FORGE;
+    } else if (handshake.getProtocolVersion().compareTo(ProtocolVersion.MINECRAFT_1_7_6) <= 0) {
+      // 1.7 Forge will not notify us during handshake. UNDETERMINED will listen for incoming
+      // forge handshake attempts. Vanilla clients will remain UNDETERMINED.
```

Comment:
You should set the connection type as `VANILLA` as soon as you are confident that this is the case, rather than leaving this as undetermined.

Reply:
Whoops that comment is wrong. It is set to `VANILLA` after the `JoinGame` packet